frontend/src/metabase/querying/notebook/components/JoinStep/JoinTablePicker/JoinTablePicker.tsx

```diff
@@ -33,6 +36,17 @@ export function JoinTablePicker({
 }: JoinTablePickerProps) {
   const isDisabled = isReadOnly;
 
+  const metadata = useSelector(getMetadata);
+  const databaseId = useMemo(() => {
+    return Lib.databaseID(query);
+  }, [query]);
+  const databases = useMemo(() => {
+    const database = metadata.database(databaseId);
+    return [database, metadata.savedQuestionsDatabase()].filter(
+      Boolean,
+    ) as Database[];
```

Comment:
`NotebookDataPicker` accepts `databases: Database[]` where `Database` comes from `metabase-types/api`.

`database` and `metadata.savedQuestionsDatabase()` (apart from being possibly `undefined`) are of type `Database` coming from `metabase-lib/v1`.


Type cast hides this problem ([see line 68](https://github.com/user-attachments/assets/966fcba8-c1ce-4ccd-b37f-208c895208ca)):

```suggestion
    return [database, metadata.savedQuestionsDatabase()].filter(isNotNull);
```

Reply:
Ahh, thanks! I assumed they were the same type since the names were the same 🤦 . Anyway, I have fixed it in https://github.com/metabase/metabase/pull/54272/commits/faf4fa5168368934e61f6825319e55fee0d266f2 that addresses @ranquild 's comment about not passing an additional prop, but constructing a `databases` variable right inside the `EmbeddingDataPicker` (aka `LegacyDataPicker`)